### ISSUE TITLE
When I changed org-roam-directory, org-roam-node-find still populates with old node names, even after sync

### ISSUE THREAD

**Issue by holtzermann17** [org-roam/org-roam#2316]
In my current setting, `org-roam-directory` is now set to `~/DirectoryB/` (let's say), whereas before it was set to `~/DirectoryA/`.

Similarly, `org-roam-db-location` is set to `~/DirectoryB/org-roam.db`.

But, when I run M-x `org-roam-node-find`, I still see some node names familiar from `~/DirectoryA/` listed (alongside) the contents of `~/DirectoryB`.

This seems to be a caching issue, because `org-roam-node-list` returns the correct files, only from the new directory.  Digging into the code, I guess I need to manually delete the contents of `org-roam-node-history` if I want the old history not to be displayed.

However, the documentation of `org-roam-node-list` — i.e., the source of the main list of options that's used to populate the interactive read in `org-roam-node-find` —  says that it is supposed to return a list of nodes *stored in the database*.  Therefore, I would expect completions based on some other database no longer to be appropriate to the new situation.

**tl;dr** I would have expected that syncing the new database via `org-roam-db-sync` to have *also* deleted the contents of `org-roam-node-history` which were related to the old database.  Otherwise, this doesn't feel like good data encapsulation.  (For comparison, if I change directories in the Emacs shell and type `ls TAB` then I am given completions related to the current directory, not some other directory.)

**Comment by nobiot** [comment]
I don't seem to be able to reproduce the issue on my end.

```
(setq org-roam-directory "~/DirectoryB/")
(org-roam-db-sync)
```
`org-roam-node-find` shows content of DirectoryB

And then I evaluate the following:
```
(setq org-roam-directory "~/DirectoryA/")
(org-roam-db-sync)
```
`org-roam-node-find` shows only content of DirectoryA. I can't find nodes in DirectoryB if I typed one.

Given this, I suspect what you see might be influenced by your completion framework (e.g. Vertico, Ivy, Helm) or the package that works with the history.

I use Vertico and built-in Savehist-mode.
